File: VB/ViewModel/EFCore/LocalData/MainViewModel.vb
Imports DevExpress.Mvvm
Imports EFCoreIssues.Issues
Imports DevExpress.Mvvm.DataAnnotations
Imports System.Linq
Imports System.Collections.Generic
Imports DevExpress.Mvvm.Xpf

Public Class MainViewModel
    Inherits ViewModelBase
    Private _Context As IssuesContext
    Private _ItemsSource As IList(Of User)
    Public ReadOnly Property ItemsSource As IList(Of User)
        Get
            If _ItemsSource Is Nothing AndAlso Not DevExpress.Mvvm.ViewModelBase.IsInDesignMode Then
                _Context = New IssuesContext()
                _ItemsSource = _Context.Users.ToList()
            End If
            Return _ItemsSource
        End Get
    End Property
    <Command>
    Public Sub ValidateRow(ByVal args As RowValidationArgs)
        Dim item = CType(args.Item, User)
        If args.IsNewItem Then _Context.Users.Add(item)
        _Context.SaveChanges()
    End Sub
    <Command>
    Public Sub ValidateRowDeletion(ByVal args As ValidateRowDeletionArgs)
        Dim item = CType(args.Items.Single(), User)
        _Context.Users.Remove(item)
        _Context.SaveChanges()
    End Sub
    <Command>
    Public Sub DataSourceRefresh(ByVal args As DataSourceRefreshArgs)
        _ItemsSource = Nothing
        _Context = Nothing
        RaisePropertyChanged(Nameof(ItemsSource))
    End Sub

End Class

File: CS/ViewModel/EFCore/LocalData/MainViewModel.cs
using DevExpress.Mvvm;
using System.Collections.ObjectModel;
using System.Linq;

namespace EFCoreIssues {
    public class MainViewModel : ViewModelBase {
        EFCoreIssues.Issues.IssuesContext _Context;
        ObservableCollection<EFCoreIssues.Issues.User> _ItemsSource;

        UserCopyOperationsSupporter _CopyOperationsSupporter;
        public UserCopyOperationsSupporter CopyOperationsSupporter {
            get {
                if(_CopyOperationsSupporter == null) {
                    _CopyOperationsSupporter = new UserCopyOperationsSupporter();
                }
                return _CopyOperationsSupporter;
            }
        }

        public ObservableCollection<EFCoreIssues.Issues.User> ItemsSource
        {
            get
            {
                if(_ItemsSource == null && !IsInDesignMode) {
                    _Context = new EFCoreIssues.Issues.IssuesContext();
                    _ItemsSource = new ObservableCollection<Issues.User>(_Context.Users);
                }
                return _ItemsSource;
            }
        }
        [DevExpress.Mvvm.DataAnnotations.Command]
        public void ValidateRow(DevExpress.Mvvm.Xpf.RowValidationArgs args) {
            var item = (EFCoreIssues.Issues.User)args.Item;
            if(args.IsNewItem)
                _Context.Users.Add(item);
            _Context.SaveChanges();
        }
        [DevExpress.Mvvm.DataAnnotations.Command]
        public void ValidateRowDeletion(DevExpress.Mvvm.Xpf.ValidateRowDeletionArgs args) {
            var item = (EFCoreIssues.Issues.User)args.Items.Single();
            _Context.Users.Remove(item);
            _Context.SaveChanges();
        }
        [DevExpress.Mvvm.DataAnnotations.Command]
        public void DataSourceRefresh(DevExpress.Mvvm.Xpf.DataSourceRefreshArgs args) {
            _ItemsSource = null;
            _Context = null;
            RaisePropertyChanged(nameof(ItemsSource));
        }
    }
}